### PR TITLE
[14.0] [FIX] purchase_allowed_product: search product by vendor reference

### PR DESCRIPTION
**Pull request by renda-dev**
Back port of https://github.com/OCA/purchase-workflow/pull/1797

### CODE REVIEW

File: purchase_allowed_product/README.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@ Purchase and Invoice Allowed Product
    !! This file is generated by oca-gen-addon-readme !!
    !! changes will be overwritten.                   !!
    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-   !! source digest: sha256:dad52af1ba2e3221412aeab43a4aac234fb45a36a9089fdf4b6f00db38f58ae9
+   !! source digest: sha256:d32423644e8e6dc1fb71183768911fb2232be0c7e2d1630a2356f526b6ac8ce5
```
Comment (HviorForgeFlow) [inline purchase_allowed_product/README.rst:10]: Can you clean up this change? thanks! @renda-dev

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

File: purchase_allowed_product/README.rst (hunk 1 above)
Comment (GSLabIt) [reply purchase_allowed_product/README.rst:10]: it is changed by ocabot afaik

**Review by GSLabIt — APPROVED** [review]

File: purchase_allowed_product/README.rst (hunk 1 above)
Comment (francesco-ooops) [reply purchase_allowed_product/README.rst:10]: @HviorForgeFlow could you update your review?

Comment (francesco-ooops) [reply purchase_allowed_product/README.rst:10]: @HviorForgeFlow gentle reminder

Comment (francesco-ooops) [reply purchase_allowed_product/README.rst:10]: @HviorForgeFlow gentle reminder

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Changing dependencies in a so old version is not advisable. Please take it into account next time, @HviorForgeFlow

**Pull request merged**